src/xcode/ENA/ENA/Source/Models/Home/HomeRiskCellConfigurator/HomeRiskCellConfigurators/HomeHighRiskCellConfigurator.swift

```diff
@@ -47,43 +51,56 @@ final class HomeHighRiskCellConfigurator: HomeRiskLevelCellConfigurator {
 
 	override func configure(cell: RiskLevelCollectionViewCell) {
-		cell.delegate = self
-
-		let title: String = isLoading ? AppStrings.Home.riskCardStatusCheckTitle : AppStrings.Home.riskCardHighTitle
-		let titleColor: UIColor = .enaColor(for: .textContrast)
-		cell.configureTitle(title: title, titleColor: titleColor)
-		cell.configureBody(text: "", bodyColor: titleColor, isHidden: true)
-
-		let color: UIColor = .enaColor(for: .riskHigh)
-		let separatorColor: UIColor = .enaColor(for: .hairlineContrast)
 		var itemCellConfigurators: [HomeRiskViewConfiguratorAny] = []
-		if isLoading {
-			let isLoadingItem = HomeRiskLoadingItemViewConfigurator(title: AppStrings.Home.riskCardStatusCheckBody, titleColor: titleColor, isLoading: true, color: color, separatorColor: separatorColor)
-			itemCellConfigurators.append(isLoadingItem)
-		} else {
-			let numberOfDaysSinceLastExposure = daysSinceLastExposure ?? 0
-			let numberContactsTitle = String(format: AppStrings.Home.riskCardHighNumberContactsItemTitle, numberRiskContacts)
-			let item1 = HomeRiskImageItemViewConfigurator(title: numberContactsTitle, titleColor: titleColor, iconImageName: "Icons_RisikoBegegnung", iconTintColor: titleColor, color: color, separatorColor: separatorColor)
-			let lastContactTitle = String(format: AppStrings.Home.riskCardLastContactItemTitle, numberOfDaysSinceLastExposure)
-			let item2 = HomeRiskImageItemViewConfigurator(title: lastContactTitle, titleColor: titleColor, iconImageName: "Icons_Calendar", iconTintColor: titleColor, color: color, separatorColor: separatorColor)
-			let dateTitle = String(format: AppStrings.Home.riskCardDateItemTitle, lastUpdateDateString)
-			let item3 = HomeRiskImageItemViewConfigurator(title: dateTitle, titleColor: titleColor, iconImageName: "Icons_Aktualisiert", iconTintColor: titleColor, color: color, separatorColor: separatorColor)
-			itemCellConfigurators.append(contentsOf: [item1, item2, item3])
+		
+		switch riskProviderState {
+		case .downloading:
+			itemCellConfigurators += [setupDownloadingCellState(for: cell)]
+		case .detecting:
+			itemCellConfigurators += [setupDetectingCellState(for: cell)]
+		default:
+			itemCellConfigurators.append(contentsOf: setupNormalCellState(for: cell))
 		}
+
 		cell.configureRiskViews(cellConfigurators: itemCellConfigurators)
 		cell.configureBackgroundColor(color: color)
-
-		let intervalTitle = String(format: AppStrings.Home.riskCardIntervalUpdateTitle, "\(detectionInterval)")
+		cell.delegate = self
+		cell.configureBody(text: "", bodyColor: titleColor, isHidden: true)
 		cell.configureDetectionIntervalLabel(
-			text: intervalTitle,
+			text: String(format: AppStrings.Home.riskCardIntervalUpdateTitle, "\(detectionInterval)"),
 			isHidden: detectionIntervalLabelHidden
 		)
 
-
 		configureButton(for: cell)
 		setupAccessibility(cell)
 	}
 
+	// MARK: - Configuration helpers.
+
+	private func setupDownloadingCellState(for cell: RiskLevelCollectionViewCell) -> HomeRiskViewConfiguratorAny {
```

Comment:
Its looks strange, that the cell is partly configured here, then another object is returned, which also configures the same cell at another place.
Maybe this can be done easier at one place?

Reply:
I fully agree, however I am currently unhappy with the entire creation and state management process for the cells. It is rather complex, error prone and difficult to adjust (some risk cells are implemented slightly different and we have a lot of duplicated code among them). I would suggest to tackle this as a separate refactoring task, what do you think?

Reply:
Agree!